src/php/Lang/Set.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Phel\Lang;
+
+use ArrayAccess;
+use Countable;
+use InvalidArgumentException;
+use Iterator;
+use Phel\Printer;
+
+class Set extends PhelArray implements ArrayAccess, Countable, Iterator, ICons, ISlice, ICdr, IRest, IPop, IRemove, IPush, IConcat
```

Comment:
Ok, I will make the required change to get to this. Btw, maybe we could use traits in order to reuse some common functions. For instance, the `hash` function is implemented in the same way all over the project. It could be nice to factorize the code, and traits are probably a better approach than inheritance. I will open an issue about it to discuss about it.